File: gcc/hls_node/fortran/hls_node.f90
!~ FAIRE APPEL A UNE FONCTION C !		
		subroutine mpc_user_main
        
        include 'mpif.h'
        
!      	Variables globales
        integer table
        save table
!~         common /comon/ table
        integer ierror
		integer rank
		
        !$hls node (table)
        
        call mpi_init(ierror)
        call mpi_comm_size(MPI_COMM_WORLD, sizes, ierror)
        call mpi_comm_rank(MPI_COMM_WORLD, rank, ierror)

        table = rank
        call mpi_barrier(MPI_COMM_WORLD, ierror)
		
		if(rank .ne. table) then
			write(6,*) 'HLS NODE FAILURE'
		endif
        
        if(rank .eq. 0) then
             write(6,*) 'HLS NODE SUCCESS'
        endif
		call mpi_finalize(ierror)
        end

File: gcc/hls_node/c/hls_node.c
#include <mpi.h>
#include <stdlib.h>
#include <stdio.h>
#include <unistd.h>

double *table;

#pragma hls node(table)

int main( int argc, char **argv )
{
	int rank, size;
	MPI_Status status;
	MPI_Request request;
	int i = 0;
	double *addr;
	
	MPI_Init( &argc, &argv );
	MPI_Comm_rank( MPI_COMM_WORLD, &rank );
	MPI_Comm_size( MPI_COMM_WORLD, &size );
	
	fprintf(stderr,"addr table = %p\n", &table);

	//node
	if(rank == 0)
	{
		for(i=1 ; i<size ; i++)
		{
			MPI_Recv(&addr, 1, MPI_DOUBLE, i, 0, MPI_COMM_WORLD, &status);

			if(addr != (double *)&table)
			{
				fprintf(stderr,"HLS NODE FAILURE\n");
				abort();
			}
		}
		fprintf(stderr,"HLS NODE SUCCESS\n");
	}
	else
	{
		double *pointeur = (double *)&table;
		MPI_Send(&pointeur, 1, MPI_DOUBLE, 0, 0, MPI_COMM_WORLD);
	}

	MPI_Finalize();
	return EXIT_SUCCESS;
}
